chain/chain/src/store.rs

```diff
@@ -2775,8 +2775,14 @@ impl<'a> ChainStoreUpdate<'a> {
                 block_hash,
             )?;
         }
+
+        // Convert trie changes to database ops for trie nodes.
+        // Create separate store update for deletions, because we want to update cache and don't want to remove nodes
+        // from the store.
+        let mut deletions_store_update = self.store().store_update();
```

Comment:
This is another quite questionable bit of code, we should think about how to make this clearer in a follow-up